packager/packager.cc

```diff
@@ -690,14 +694,41 @@ FullyQualifiedName getFullyQualifiedName(core::Context ctx, const ast::Unresolve
     return fqn;
 }
 
-// Gets the package name in `tree` if applicable.
-PackageName getPackageName(core::Context ctx, const ast::UnresolvedConstantLit *constantLit) {
+PackageName getPackageName(core::Context ctx, const ast::UnresolvedConstantLit *constantLit,
+                           core::ClassOrModuleRef symbol) {
     ENFORCE(constantLit != nullptr);
 
     PackageName pName;
     pName.fullName = getFullyQualifiedName(ctx, constantLit);
 
     // pname.mangledName will be populated later, when we have a mutable GlobalState
+    pName.mangledName.owner = symbol;
+
+    return pName;
+}
+
+PackageName getUnresolvedPackageName(core::Context ctx, const ast::UnresolvedConstantLit *constantLit) {
+    ENFORCE(constantLit != nullptr);
+
+    PackageName pName;
+    pName.fullName = getFullyQualifiedName(ctx, constantLit);
+
+    // Since packager now runs after namer, we know that these symbols are entered.
+    auto owner = core::Symbols::PackageSpecRegistry();
+    for (auto part : pName.fullName.parts) {
+        auto member = owner.data(ctx)->findMember(ctx, part);
+        if (!member.exists() || !member.isClassOrModule()) {
+            owner = core::Symbols::noClassOrModule();
+            break;
+        }
+        owner = member.asClassOrModuleRef();
+    }
+
+    if (owner == core::Symbols::PackageSpecRegistry()) {
+        owner = core::Symbols::noClassOrModule();
+    }
```

Comment:
Would you mind adding a comment to the effect of this case only being possible if the full name parts are empty? (At least, I think that's the case...)

Reply:
I'm happy to. The reason I didn't is because I don't expect this code to be long-lived.

I'm using this `owner` field as a proxy for what will eventually be a `PackageRef` symbol. In the future, namer will create Package symbols and we will be able to lookup a `PackageRef` for a given `ClassOrModuleRef` directly in packager.cc, such that we will either have `pkg.exists()` or `core::Symbols::noPackage()` here.

Reply:
(So I'll add the comment, but I expect it to be shortlived.)